### DISCUSSION TITLE
How to use SRT in TTS?

### DISCUSSION THREAD

**Discussion by vshlykov** [coqui-ai/TTS#3357] [General]
Hello!

I have subtitles file with timings, like this:

[00:0.000 --> 00:7.500] Hello! Let's try to play Xenonauts 2 again today, because the 1.5 gig patch has arrived.
[00:7.500 --> 00:14.200] I think when I downloaded the game, it weighed about 2 gigs, so there's a lot of rework along the way.

Is it possible to make it to speech with pauses between sentenses equal time on string start?

**Comment by seldem** [comment]
I have a similar case, it should skip the sequence numbers and should obey the timings between sentences. I could not find a solution yet.

16
00:01:24,839 --> 00:01:32,360
You will not be able to work with thicker threads, such as sewing thread or fishing line.
 
17
00:01:32,360 --> 00:01:37,880
Because we go through our beads many times and we have to work hard

**Reply by vshlykov** [reply]
I use this one: https://github.com/bnsantoso/sub-to-audio

**Reply by TalonPower** [reply]
do you have any clue how to use this with the colab?  no instructions cant find a tutorial on how to use it anywhere

**Reply by vshlykov** [reply]
I don't use colab. After install TTS and subtoaudio I used this code:

```
from TTS.api import TTS
from subtoaudio import SubToAudio

sub = SubToAudio(model_name="tts_models/multilingual/multi-dataset/xtts_v2")
subtitle = sub.subtitle("filtered_subtitles.srt")
sub.convert_to_audio(sub_data=subtitle, output_path="subtitle11.wav", language="en", speaker_wav="speach-eng-letter.wav", tempo_mode="overflow", tempo_limit=1.2, shift_mode="right")
```

**Reply by madey83** [reply]
> I don't use colab. After install TTS and subtoaudio I used this code:
> 
> ```
> from TTS.api import TTS
> from subtoaudio import SubToAudio
> 
> sub = SubToAudio(model_name="tts_models/multilingual/multi-dataset/xtts_v2")
> subtitle = sub.subtitle("filtered_subtitles.srt")
> sub.convert_to_audio(sub_data=subtitle, output_path="subtitle11.wav", language="en", speaker_wav="speach-eng-letter.wav", tempo_mode="overflow", tempo_limit=1.2, shift_mode="right")
> ```


hello vshlykov

could you please help me with execute your command with RTX GPU instead of CPU?

**Reply by vshlykov** [reply]
My code works with GPU.

**Reply by madey83** [reply]
> My code works with GPU.

i have copied your code and executed it and this is what i got

![image](https://github.com/user-attachments/assets/8a79fd61-ec7a-4ade-8600-f72cbbd5b058)

it shows CPU usage...


could you guide me what i need to check/change to use GPU?

**Reply by madey83** [reply]
i could handle this by import pytorch
![image](https://github.com/user-attachments/assets/6bf886f5-004d-4cbf-b2da-7bf5bfe7fec1)


Could you advise how to handle special characters like ?, !  --when sentence has it voice slows down and next voice is shifted....

**Reply by vshlykov** [reply]
> > My code works with GPU.
> 
> i have copied your code and executed it and this is what i got
> it shows CPU usage...
> 
> could you guide me what i need to check/change to use GPU?

Looks like you have problem with TTS. It has to use GPU by default. Can you run default TTS without SubToAudio to check if it works with GPU?

**Reply by madey83** [reply]
it works.

when you create voice do you remove from sub file special characters?

**Reply by vshlykov** [reply]
Right now I use CapCut software.

**Reply by madey83** [reply]
> CapCut software

Have you got other free open source TTS projects similar to  Couqi-TTS that i could use...


in Couqi i do not an option to voice pitch, etc..

Many thanks for your answers.

**Reply by vshlykov** [reply]
No. I use desktop software.

**Comment by seldem** [comment]
@vshlykov Thank you for the suggestion.
